app/src/pages/portal/ManageUsers.vue

```diff
@@ -0,0 +1,301 @@
+      this.toggleDialogBox()
+    },
+    closeDialogBox () {
+      this.toggleDialogBox()
+    },
+    showUpdateForm () {
+      this.updateMode = true
+      this.selectedUser = { ...this.selected[0] }
+    },
+    closeUpdateForm () {
+      this.updateMode = false
+      this.selectedUser = {}
+      this.selected = []
+    },
+    getClass ({ _id }) {
+      return _id === this.selected?._id ? 'u--bg' : ''
+    },
+    getAlternateLabel (count) {
+      let plural = ''
+      if (count > 1) {
+        plural = 's'
+      }
+      return `${count} user${plural} selected`
+    },
+    async searchOnTable () {
+      this.pageNumber = 1
+      await this.$apollo.queries.users.refetch()
+      this.searched = this.users.data
+    },
+    onSelect (items) {
+      this.selected = items
+    },
+    async loadPrevNextUsers (event) {
+      this.pageNumber = event
+      await this.$apollo.queries.users.refetch()
+    },
+
+    async deleteUsers () {
+      const id = []
+      for (let i = 0; i < this.selected.length; i++) {
+        id.push(this.selected[i]._id)
+      }
+      if (id.includes(this.userId)) {
+        return this.$store.commit('setSnackbar', {
+          message: "Can't Delete Current User",
+          duration: 3000
+        })
+      }
+      this.loading = true
+      try {
+        await this.$apollo.mutate({
+          mutation: DELETE_USERS_QUERY,
+          variables: {
+            input: {
+              ids: id
+            }
+          }
+        })
+        this.$store.commit('setSnackbar', {
+          message: 'Users deleted successfully',
+          duration: 3000
+        })
+        await this.$apollo.queries.users.refetch()
+      } catch (error) {
+        this.$store.commit('setSnackbar', {
+          message: 'Something went wrong',
+          action: () => this.deleteUsers()
+        })
+      } finally {
+        this.loading = false
+      }
+    },
+
+    async updateUser () {
+      this.loading = true
+      this.loadingText = 'Updating User'
+      const data = this.selectedUser
+
+      try {
+        await this.$apollo.mutate({
+          mutation: UPDATE_USER_QUERY,
+          variables: {
+            input: {
+              _id: data._id,
+              givenName: data.givenName,
+              surName: data.surName,
+              roles: this.roles
+            }
+          }
+        })
+        this.$store.commit('setSnackbar', {
+          message: 'User Updated successfully',
+          duration: 3000
+        })
+        await this.$apollo.queries.users.refetch()
+        this.closeUpdateForm()
+      } catch (error) {
+        this.$store.commit('setSnackbar', {
+          message: 'Something went wrong',
+          action: () => this.updateUser()
+        })
+      } finally {
+        this.loading = false
+        this.loadingText = 'Loading'
+      }
+    }
+  },
+  created () {
+    this.$store.commit('setAppHeaderInfo', { icon: '', name: '' })
+  },
+  apollo: {
```

Comment:
For some reason whenever I route to this page locally, the users query returns a `200 response` but also gives this error in snackbar: `"INTERNAL_SERVER_ERROR": "An error occurred."`

Reply:
That is happening because I failed to add backward compatibility logic. I just added it